packages/flutter/lib/src/widgets/debug.dart

```diff
@@ -191,3 +191,20 @@ bool debugAssertAllWidgetVarsUnset(String reason) {
   });
   return true;
 }
+
+/// Holds data used by debug tools to trace execution
+class DebugTraceData {
+  /// Holds where an entity was created from
+  final SourceLocation ctorCallsite;
+
+  DebugTraceData({this.ctorCallsite});
```

Comment:
also, please add much more colour to the dartdocs... e.g. what sets the information, how do you read it, when is it available, what tools are relevant, etc. As much as possible, dartdocs should have crossreferences to other dartdocs so that when reading the docs you can have breadcrumbs to follow to learn more.